moban/mobanfile/targets.py

```diff
@@ -34,6 +37,23 @@ def _handle_explicit_target(options, target):
     data_file = target.get(constants.LABEL_CONFIG, common_data_file)
     output = target[constants.LABEL_OUTPUT]
     template_type = target.get(constants.LABEL_TEMPLATE_TYPE)
+    if template_type and len(template_type) > 0:
+        if constants.TEMPLATE_TYPES_BASE_TYPE in template_type[0]:
+            adhoc_type = uuid.uuid4().hex
+            file_extension = uuid.uuid4().hex
+            base_type = template_type[0][constants.TEMPLATE_TYPES_BASE_TYPE]
```

Comment:
this code here raise the question why an index is required. so I went back to check the syntax of `template_types`. Yet again, it is my bad. I will correct it in #202, and will post it here:

Since the user defined syntax is like the following,
```
template_types:
    my_own_type:
      base_type: jinja2
      file_extensions:
        - file_type_of_my_choice
      options:
        extensions:
          - jinja2_time.TimeExtension
```

it is preferred to have:

Since
```
  - template: a.template.that_enjoy_ad_hoc_type
    output: b.output
    template_type:
      base_type: jinja2
      file_extensions:
        - file_type_of_my_choice
      options:
        extensions:
          - jinja2_time.TimeExtension
```

with consistency, moban user needs less memory to remember things. I realized how much confused I have caused, with my original requirement. I should have double checked before confirm it.

Reply:
With the syntax being updated, this block of code will be simplified further.

Reply:
Will update.